extensions/lifecycle/LifecycleQueuePopulator.js

```diff
@@ -181,23 +218,38 @@ class LifecycleQueuePopulator extends QueuePopulatorExtension {
             if (err) {
                 this.log.error('unable to get account', {
                     ownerId,
+                    err,
                 });
+                return;
             }
-            this.log.trace('publishing bucket replication entry', { bucket: entry.bucket });
 
-            const topic = `cold-restore-req-${locationName}`;
-            this.publish(
-                topic,
-                `${entry.bucket}/${entry.key}`,
-                JSON.stringify({
-                    bucketName: entry.bucket,
-                    objectKey: entry.key,
-                    objectVersion: value.versionId,
-                    archiveInfo: value.archive.archiveInfo,
-                    requestId: uuid(),
-                    accountId
-                }),
+            this.log.trace(
+                'publishing object restore entry',
+                { bucket: entry.bucket, key: entry.key, version: value.versionId },
             );
+
+            const topic = coldStorageRestoreTopicPrefix + locationName;
+            const key = `${entry.bucket}/${entry.key}`;
+            const message = JSON.stringify({
+                bucketName: entry.bucket,
+                objectKey: entry.key,
+                objectVersion: value.versionId,
+                archiveInfo: value.archive.archiveInfo,
+                requestId: uuid(),
+                accountId
+            });
+
+            this._setupProducer(topic, producer => {
+                const kafkaEntry = { key: encodeURIComponent(key), message };
+                producer.send([kafkaEntry], err => {
+                    if (err) {
+                        this.log.error('error publishing GC.deleteData entry', {
```

Comment:
Is it not the wrong log error message?

Reply:
@nicolas2bert @RachedBenMustapha 
should a ticket be created for this to be fixed in a different PR? this will unblock a few other PRs

Reply:
@alexanderchan-scality I think we can push a change and merge this one.